apps/dashboard/config/configuration_singleton.rb

```diff
@@ -278,6 +278,16 @@ def pinned_apps_menu_length
     config.fetch(:pinned_apps_menu_length, 6)
   end
 
+  # What to group pinned apps by
+  def pinned_apps_group_by
+    group_by = config.fetch(:pinned_apps_group_by, "")
+    if group_by == 'category' || group_by == 'subcategory'
```

Comment:
Add a big `#HACK: ` comment above this explaining why we have to do "original_". Ideally the impl of this method would just be  `config.fetch(:pinned_apps_group_by, "")` but I understand why this is the way it is - because we are passing it as an argument to groups_for and the app's that we are dealing with happen to have their category and subcategory overridden for the sake of grouping those in a way that utilizes the existing menu code.

Its a code smell though because why should `ConfigurationSingleton` know about the implementation of the `FeaturedApp` object.

Maybe a better design in the future would obviate the need for FeaturedApp class to exist, or if it does, to override category and subcategory. But until then a good comment would do.

Reply:
Done, though I used `#FIXME` just so we have one thing to grep for (even though we also have `TODO`s).